patches/minecraft/net/minecraft/data/registries/RegistryPatchGenerator.java.patch

```diff
@@ -0,0 +1,13 @@
+--- a/net/minecraft/data/registries/RegistryPatchGenerator.java
++++ b/net/minecraft/data/registries/RegistryPatchGenerator.java
+@@ -17,9 +_,7 @@
+       return p_310881_.thenApply((p_309945_) -> {
+          RegistryAccess.Frozen registryaccess$frozen = RegistryAccess.fromRegistryOfRegistries(BuiltInRegistries.REGISTRY);
+          Cloner.Factory cloner$factory = new Cloner.Factory();
+-         RegistryDataLoader.WORLDGEN_REGISTRIES.forEach((p_313050_) -> {
+-            p_313050_.runWithArguments(cloner$factory::addCodec);
+-         });
++         net.minecraftforge.registries.DataPackRegistriesHooks.getDataPackRegistriesWithDimensions().forEach(registryData -> registryData.runWithArguments(cloner$factory::addCodec));
```

Comment:
This is wrong, you're changing WORLDGEN_REGISTRIES to DIMENSION_REGISTRIES
Also this can be a 1 line patch if you don't rename/inline the lambda

Reply:
Does `DataPackRegistriesHooks.getDataPackRegistriesWithDimensions()` not include the worldgen registries with the dimension registries?

Reply:
pretty sure it doesnt. as I references a different field as I stated in my comment

Reply:
what a shitty method name, then

Reply:
yarp

Reply:
Sorry Lex, you are wrong, `DataPackRegistriesHooks#getDataPackRegistriesWithDimensions` returns a Stream concat of `DataPackRegistriesHooks#getDataPackRegistries` including the worldgen registries and the dimensions registries.

Reply:
just cross referenced, this is true.
https://github.com/MinecraftForge/MinecraftForge/blob/5d0047753dfac0caaf5d97cc3f5c9a8b0990cb44/src/main/java/net/minecraftforge/registries/DataPackRegistriesHooks.java#L62-L64

This comment thread is not resolved; I am still in agreement that the patch itself should be simplified.

Reply:
Just to clarify, ya I was wrong. The DataPackRegistriesHooks.DATA_PACK_REGISTRIES_VIEW is a view of 
 RegistryDataLoader.WORLDGEN_REGISTRIES. my proposal is to bypass the stupidly named Forge class and just do a one line patch that makes things clear:
 ```diff
          RegistryDataLoader.WORLDGEN_REGISTRIES.forEach((p_313050_) -> {
             p_313050_.runWithArguments(cloner$factory::addCodec);
          });
+        RegistryDataLoader.DIMENSION_REGISTRIES.forEach(reg -> reg.runWithArguments(cloner$factory::addCodec)
 ```